src/pages/workspaces/workspace/workflows/DataStepContent.js

```diff
@@ -41,10 +41,12 @@ export default class DataStepContent extends Component {
   isValidSelectionModel() {
     const { entitySelectionModel } = this.state
     const { newSetName, selectedEntities, type } = entitySelectionModel
-    const { entityType, name } = selectedEntities
+    const selectionSize = _.size(selectedEntities)
+
     return (type === EntitySelectionType.processAll ||
-      (type === EntitySelectionType.processFromSet && !!entityType && !!name) ||
-      (_.size(selectedEntities) > 0 && !!newSetName))
+      (type === EntitySelectionType.chooseRows && (selectionSize === 1 || (selectionSize > 1 && !!newSetName))) ||
+      (type === EntitySelectionType.chooseSet && (selectionSize === 1 || (selectionSize > 1 && selectionSize <= 10 && !!newSetName)))
```

Comment:
Currently the `processFromSet` is always invalid. There should be a case for that, which I think should validate `newSetName` instead of this line.

Reply:
I _think_ I fixed this?